pandas/core/groupby/groupby.py

```diff
@@ -950,7 +951,18 @@ def curried(x):
             if name in base.plotting_methods:
                 return self.apply(curried)
 
-            return self._python_apply_general(curried, self._obj_with_exclusions)
+            result = self._python_apply_general(curried, self._obj_with_exclusions)
+
+            if result.ndim == 1 and self.obj.ndim == 1 and result.name != self.obj.name:
```

Comment:
`result.name != self.obj.name` will be wrong with np.nan (and will raise with pd.NA)

Reply:
Doh, thanks. This highlights the fact that we shouldn't be fixing this here, but rather the logic that's using in wrapping the apply results. I didn't want this PR spreading out to that method though. I've opened #46369 for this; I'm thinking here we ignore testing the name, and fix properly.